epo_ops/api.py

```diff
@@ -35,13 +37,52 @@ def __init__(self, key, secret, accept_type="xml", middlewares=None):
         self.secret = secret
         self._access_token = None
 
-    def family(self, reference_type, input, endpoint=None, constituents=None):
+    def family(
+        self,
+        reference_type: str,
+        input: Union[Docdb, Epodoc],
+        endpoint=None,
+        constituents: Optional[List[str]] = None,
+    ) -> requests.Response:
+        """
+        Retrieves the patent numbers of the extended patent family related to the input (INPADOC family).
+
+        Args:
+            reference_type (str): Any of "publication", "application", or "priority".
+            input (Epodoc or Docdb): The document number. Cannot be Original.
+            endpoint (optional): None. Not applicable for family service.
+            constituents (list[str], optional): List of "biblio", "legal" or both.
+                                                Defaults to None.
+
+        Returns:
+            requests.Response: a requests.Response object.
+
+        Examples:
+            >>> response = client.family("publication", epo_ops.models.Epodoc("EP1000000"))
+            >>> response
+            <Response [200]>
+            >>> len(response.text)
+            8790
+
+            >>> response_with_constituents = client.family("publication", epo_ops.models.Epodoc("EP1000000"), None, ["biblio", "legal"])
+            >>> response_with_constituents
+            <Response [200]>
+            >>> len(response_with_constituents.text)
+            160206
+        """
+        if endpoint is not None:
+            warnings.warn(
+                "The `endpoint` argument is not used in this context and will be removed.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+
         url = self._make_request_url(
             dict(
                 service=self.__family_path__,
                 reference_type=reference_type,
                 input=input,
-                endpoint=endpoint,
+                endpoint=None,
```

Comment:
If we are 100% sure on this, let's do it. Otherwise, it would also be a breaking change, if, by chance, the parameter would still be needed in some cases.

Reply:
I tested the family() method, and the implementation allows `endpoint` to be assigned one of the allowed values for the parameter `constituents` and still work as if `constituents` was assigned the value. Although not according to ops specs, the resulting url when using e.g. 'endpoint="biblio" is the same as if `constituents=["biblio"]` was used.
So removing it may indeed break code out there. I'd leave that for a major version upgrade. Fine with the deprecation warning.